common/Model/Api.php

```diff
@@ -281,25 +289,53 @@ public function sendSyncUrl(string $syncUrl, ?int $storeId): void {
         ];
 
         try {
-            $this->doSendSyncUrl($url, $data);
+            $this->doRequest($url, 'POST', $data);
         } catch (\Exception $e) {
             $this->logger->error(sprintf('Sending sync URL to Dashboard failed with error %s', $e->getMessage()));
         }
     }
 
-    private function doSendSyncUrl(string $url, array $data): void {
+    private function hasConsent(string $order_number, array $config): bool {
+        $url =  sprintf(
+            self::ORDER_CONSENT_URL,
+            $this->extension->getDashboardDomain(),
+            $config['webshop_id'],
+            $config['api_key'],
+            $order_number
+        );
+
+        try {
+            $response_data = $this->doRequest($url, 'GET',null);
+        } catch (\Exception $e) {
+            $this->logger->error(sprintf(
+                'Checking order (%s consent failed with error %s',
+                $order_number,
+                $e->getMessage()
+            ));
+            return false;
+        }
+
+        return $response_data['has_consent'] ?? false;
+    }
+
+
+    private function doRequest(string $url ,string $method, ?array $data ): ?array {
         $curl = curl_init();
```

Comment:
You should reuse the cURL handle so as not to make two connections for each invitation (one for consent check, one for sending the invite).